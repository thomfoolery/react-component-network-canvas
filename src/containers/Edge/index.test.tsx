import React from "react";
import renderer from "react-test-renderer";
import {
  mockDragManager,
  mockGraphManager,
  mockWorkspace,
  mockBridge,
} from "@component/utils/mocks";
import {
  useDragManager,
  useGraphManager,
  useWorkspace,
  useBridge,
} from "@component/hooks";

import {Edge} from "./index";

jest.mock("@component/hooks", () => ({
  useDragManager: jest.fn(),
  useGraphManager: jest.fn(),
  useWorkspace: jest.fn(),
  useBridge: jest.fn(),
}));

useDragManager.mockImplementation(() => mockDragManager());
useGraphManager.mockImplementation(() => mockGraphManager());
useWorkspace.mockImplementation(() => mockWorkspace());
useBridge.mockImplementation(() => mockBridge());

describe("Edge", () => {
  it("renders correctly", () => {
    const key = "edge-1";
    const edge = {
      id: "edge-1",
      from: {
        nodeId: "node-1",
        portId: "node-1-output-1",
      },
      to: {
        nodeId: "node-2",
        portId: "node-2-input-1",
      },
    };
    const isDraft = false;
    const tree = renderer
      .create(<Edge key={key} edge={edge} isDraft={isDraft} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
