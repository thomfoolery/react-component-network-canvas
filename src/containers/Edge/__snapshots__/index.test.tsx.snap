// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edge renders correctly 1`] = `
<g
  data-testid="edge-1"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <path
    className="Edge"
    id="Edge-edge-1"
  />
  <path
    className="EdgeArea"
  />
</g>
`;
